src/Adapter/Order/CommandHandler/AbstractOrderCommandHandler.php

```diff
@@ -0,0 +1,174 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Order\CommandHandler;
+
+use Configuration;
+use Context;
+use OrderDetail;
+use Pack;
+use PrestaShop\PrestaShop\Adapter\Order\AbstractOrderHandler;
+use PrestaShop\PrestaShop\Adapter\StockManager;
+use PrestaShop\PrestaShop\Core\Domain\Order\Exception\OrderException;
+use Product;
+use StockAvailable;
+use StockManagerFactory;
+use StockMvt;
+use Warehouse;
+
+/**
+ * Abstracts reusable functionality for Order subdomain handlers.
+ *
+ * @internal
+ */
+abstract class AbstractOrderCommandHandler extends AbstractOrderHandler
+{
+    /**
+     * @param OrderDetail $orderDetail
+     * @param int $productQuantity
+     * @param bool $delete
+     */
+    protected function reinjectQuantity(OrderDetail $orderDetail, $productQuantity, $delete = false)
```

Comment:
Does this function really beong to an OrderCommandHandler ? It rather seems to be a generic Product / Stock processing.

Almost feels like a ProductCommandHandler or a ProductCommanderTrait 🤔 .

Again we are walking on the boundaries between domains 😅

Reply:
Hmm, I think it's part of `Order` domain. It was used only in `AdminOrdersController` before.

> Does this function really beong to an OrderCommandHandler ? It rather seems to be a generic Product / Stock processing.

It restocks products **only from Order** by using `OrderDetail`, so it's related to `Orders` more or less I think.

Reply:
👍